File: src/nl/rubensten/texifyidea/structure/latex/LatexSubSubSectionPresentation.java
package nl.rubensten.texifyidea.structure.latex;

import nl.rubensten.texifyidea.TexifyIcons;
import nl.rubensten.texifyidea.psi.LatexCommands;
import nl.rubensten.texifyidea.structure.EditableHintPresentation;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

/**
 * @author Ruben Schellekens
 */
public class LatexSubSubSectionPresentation implements EditableHintPresentation {

    private final String subSubSectionName;
    private String hint = "";

    public LatexSubSubSectionPresentation(LatexCommands sectionCommand) {
        if (!sectionCommand.getCommandToken().getText().equals("\\subsubsection")) {
            throw new IllegalArgumentException("command is no \\subsubsection-command");
        }

        this.subSubSectionName = sectionCommand.getRequiredParameters().get(0);
    }

    @Nullable
    @Override
    public String getPresentableText() {
        return subSubSectionName;
    }

    @Nullable
    @Override
    public String getLocationString() {
        return hint;
    }

    @Nullable
    @Override
    public Icon getIcon(boolean b) {
        return TexifyIcons.DOT_SUBSUBSECTION;
    }

    @Override
    public void setHint(@NotNull String hint) {
        this.hint = hint;
    }
}

File: src/nl/rubensten/texifyidea/structure/latex/LatexSubSubSectionPresentation.kt
package nl.rubensten.texifyidea.structure.latex

import nl.rubensten.texifyidea.TexifyIcons
import nl.rubensten.texifyidea.psi.LatexCommands
import nl.rubensten.texifyidea.structure.EditableHintPresentation

/**
 * @author Ruben Schellekens
 */
class LatexSubSubSectionPresentation(sectionCommand: LatexCommands) : EditableHintPresentation {

    private val subSubSectionName: String
    private var hint = ""

    init {
        if (sectionCommand.commandToken.text != "\\subsubsection") {
            throw IllegalArgumentException("command is no \\subsubsection-command")
        }

        this.subSubSectionName = sectionCommand.requiredParameters[0]
    }

    override fun getPresentableText() = subSubSectionName

    override fun getLocationString() = hint

    override fun getIcon(b: Boolean) = TexifyIcons.DOT_SUBSUBSECTION!!

    override fun setHint(hint: String) {
        this.hint = hint
    }
}
